docs/application/dotnet/guides/nui/text.md

```diff
@@ -0,0 +1,578 @@
+    ```
+
+    **Figure: Text with drop shadow (top), bigger shadow (middle), and color shadow (bottom)**
+
+    ![Text with drop shadow](media/TextWithShadow.png)
+
+    ![Text with bigger shadow](media/TextWithBiggerShadow.png)
+
+    ![Text with color shadow](media/TextWithColorShadow.png)
+
+    Shadow parameters can also be set using a JSON string.
+
+-   To underline the text label, set the `Underline` property:
+
+    ```csharp
+    label1.Text = "Text with Underline";
+
+    PropertyMap textStyle = new PropertyMap();
+    textStyle.Add("enable", new PropertyValue("true"));
+    label1.Underline = textStyle;
+    ```
+
+    By default, the underline height is based on the font metrics. For example, the following text figures are in one pixel height:
+
+    **Figure: Text with underline**
+
+    ![Text with underline](media/TextWithUnderline.png)
+
+    You can set the underline color and height using a property map:
+
+    ```csharp
+    label2.Text = "Text with Color Underline";
+
+    PropertyMap textStyle = new PropertyMap();
+    textStyle.Add("enable", new PropertyValue("true"));
+    textStyle.Add("color", new PropertyValue(Color.Green));
+    textStyle.Add("height", new PropertyValue(2.0f)); /// 2-pixel height
+    label2.Underline = textStyle;
+    ```
+
+    **Figure: Text with color underline**
+
+    ![Text with color underline](media/TextWithColorUnderline.png)
+
+
+<a name="scrolling"></a>
+-   To enable text scrolling, set the `EnableAutoScroll` property to `true`:
+
+    ```csharp
+    label.EnableAutoScroll = true;
+    ```
+
+    After scrolling is enabled, scrolling continues until the loop count is reached or `EnableAutoScroll` is set to `false`. When `EnableAutoScroll` is set to `false`, the text completes its current scrolling loop before it stops scrolling.
+
+    **Figure: Auto-scrolling text**
+
+    ![Auto-scrolling text](media/AutoScroll.gif)
+
+    Auto-scrolling enables text to scroll within the text table. You can use it to show the full content if the text exceeds the boundary of the control. You can also scroll text that is smaller than the control. To ensure that the same part of the text is not visible in more than one place at the same time, you can configure the gap between repetitions. The left-to-right text always scrolls left and the right-to-left text always scrolls right.
+
+    You can set the scroll speed, gap, and loop count in the stylesheet or using the following properties:
+
+    -   `AutoScrollSpeed` property defines the scrolling speed in pixels/second.
+    -   `AutoScrollLoopCount` property specifies how many times the text completes a full scroll cycle. For example, if this property is set to 3, the text scrolls across the control three times and then stops. If this property is set to 0, scrolling continues until `EnableAutoScroll` is set to false.
+
+        Setting `EnableAutoScroll` to `false` stops scrolling while maintaining the original loop count value for the next start.

```

Comment:
Also, I suggest keeping line 289 in list like this: 

-   `AutoScrollSpeed` property defines the scrolling speed in pixels per second.
-   `AutoScrollLoopCount` property specifies the number of times the text completes a full scroll cycle. For example, if this property is set to `3`, the text scrolls across the control three times and then stops. If this property is set to `0`, the text scroll continues until `EnableAutoScroll` is set to `false`.

       - Setting `EnableAutoScroll` to `false` stops scrolling while maintaining the original loop count value for the next start. 

-   `AutoScrollGap` property specifies the amount of white space in pixels. The white space gets displayed before the scrolling text appears again. This gap automatically increases, if the given value is not large enough to prevent the same part of the text from appearing twice at the same time.

Ignore if you don't like.